XMonad/Actions/Menu/KeyParse.hs

```diff
@@ -0,0 +1,347 @@
+module XMonad.Actions.Menu.KeyParse where
+
+{-|
+
+Module      : XMonad.Menu.Actions.KeyParse
+Description : A set of tools to run and render interactive menus
+
+Maintainer  : David Janssen <janssen.dhj@gmail.com>
+Stability   : unstable
+Portability : unportable
+
+This code is taken almost entirely from the key-string parsing capabilities of
+EZConfig, written by Brent Yorgey. EZConfig does not export any of these
+functions and I did not want to modify EZConfig itself, so I extracted the code,
```

Comment:
I thought it over for a bit over Christmas, (Merry Christmas! btw :-)).

I looked over all the different modules in xmonad-contrib, and found nearly 20 different one that do something with KeySyms, KeyMasks, etc. Furthermore, there were 3 or 4 slightly different Pretty-Printers for keys, a few 'key-grabbers', and a lot of mappings from keys to usually actions. All of this spread across all the directories of xmonad-contrib.

In my perspective (but feel free to disagree) this seems to indicate that a new module (or set of modules) that unify and ease the use of working with user input through the keyboard wouldn't be uncalled for. Perhaps simply making a new subdirectory "Keys" inside XMonad and populating this with a number of modules, one of which could be the parsing capabilities from "EZConfig.hs", and then simply importing the necessary functions into EZConfig?

I'd be happy to have a go at this refactoring.